src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,982 @@
+# coding=utf-8
+# Copyright 2021 Facebook AI Research The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch MaskFormer model."""
+
+from __future__ import annotations
+
+import logging
+import math
+from dataclasses import dataclass, field
+from numbers import Number
+from typing import Any, Callable, Dict, List, Optional, Tuple, TypedDict, Union
+
+import numpy as np
+import torch
+import torchvision
+from torch import Tensor, nn
+from torch.nn import functional as F
+
+from einops import rearrange
+from einops.einops import repeat
+from timm.models.layers import DropPath, to_2tuple, trunc_normal_
+
+from ...file_utils import (
+    ModelOutput,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    is_scipy_available,
+    is_timm_available,
+    is_vision_available,
+    replace_return_docstrings,
+    requires_backends,
+)
+from ...modeling_utils import PreTrainedModel
+from ...utils import logging
+from ..detr import DetrConfig
+from ..detr.modeling_detr import DetrDecoder, DetrDecoderOutput
+from ..swin import SwinConfig, SwinModel
+from .configuration_maskformer import ClassSpec, MaskFormerConfig
+
+
+logger = logging.get_logger(__name__)
+import torch.distributed as dist
+
+
+_CONFIG_FOR_DOC = "MaskFormerConfig"
+
+PREDICTIONS_MASKS_KEY = "preds_masks"
+PREDICTIONS_LOGITS_KEY = "preds_logits"
+TARGETS_MASKS_KEY = "pixel"
+TARGETS_LABELS_KEY = "classes"
+
+# TODO this has to go away!
+from detectron2.utils.comm import get_world_size
+from scipy.optimize import linear_sum_assignment
+
+
+@dataclass
+class MaskFormerOutput(ModelOutput):
+    preds_logits: torch.FloatTensor
```

Comment:
Just `logits` for consistency with all other model outputs, since there is only one kind of logits.